api/modules/GithubEvents/index.js

```diff
@@ -80,13 +84,23 @@ class GithubEvents extends EventEmitter {
         );
     }
 
-    emitFromReq(req) {
+    async emitFromReq(req) {
+        const handlerPromises = [];
+
         Object.keys(eventsConfig).forEach(eventName => {
             if (GithubEvents.match(req, eventName)) {
-                logger.info('Request matches eventName', { req, eventName });
-                this.emit(eventName, req);
+                logger.info('Request matches eventName', { eventName });
+                this.emit(eventName, req, handlerPromises);
             }
         });
+
+        // wait for all handlers to finish, and throw if any rejections
+        await Promise.all(handlerPromises);
```

Comment:
❓ Any value in this being `Promise.allSettled` to optimistically emit as many events as it can, then inspect for any errors and log and/or throw?

Reply:
That may allow you to capture the rejections in a more straight forward way in the code by just filtering the results and may not need the use of `captureRejectionSymbol`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
Good idea. This is a good spot to aggregate and log rejections. And actually the `captureRejectionSymbol` isn't needed anymore, since I changed `ghEvents.js` to not async functions as handlers.